src/components/BShuttle/BShuttle.vue

```diff
@@ -0,0 +1,480 @@
+<template>
+  <div class='shuttle'>
+
+    <div class='available'>
+      <label :id="'available-label-' + id">{{ availableLabel }}</label>
+      <ul
+        aria-labeledby='available-label'
+        role='selectbox'
+        tabindex='0'
+      >
+        <li
+          v-if="placeholder && (!options || options.length === 0)"
+          :aria-label="placeholder"
+          class='placeholder'
+        >
+        {{ placeholder }}
+        </li>
+        <li
+          v-for="option in availableOptions"
+          :key="'available-option-' + id + '-' + option.id"
+          :id="'available-option-' + id + '-' + option.id"
+          :class="optionsClass(option, 'available')"
+          @click="onOptionClick(option, 'available', 'chosen', $event)"
+          @keyup.space.prevent.stop="onOptionClick(option, 'available', 'chosen', $event)"
+          @keyup.down.prevent.stop="onKeyupDown($event, option, 'available', 'chosen')"
+          @keyup.up.prevent.stop="onKeyupUp($event, option, 'available', 'chosen')"
+          @keyup.right.prevent.stop="onMoveSelectedClick('available')"
+          @keyup.enter.prevent.stop="onMoveSelectedClick('available')"
+          @keydown.up.prevent.stop
+          @keydown.down.prevent.stop
+          @keydown.space.prevent.stop
+          role='option'
+          tabindex='0'
+        >
+        {{ option.displayText }}
+        </li>
+      </ul>
+    </div>
+
+    <div class='buttons'>
+      <button 
+        @click="onMoveAllOptions('available')"
+        id='all-to-chosen'
+        type='button'
+        >
+        <img class="icon" :src="allToChosen" />
+      </button>
+      <button 
+        @click="onMoveSelectedClick('available')"
+        id='selected-to-chosen'
+        type='button'
+        >
+        <i class='icon ion-arrow-right-c' aria-hidden='true'></i>
+      </button>
+      <button
+        @click="onMoveSelectedClick('chosen')"
+        id='selected-to-available'
+        type='button'>
+        <i class='icon ion-arrow-left-c' aria-hidden='true'></i>
+      </button>
+      <button 
+        @click="onMoveAllOptions('chosen')"
+        id='all-to-available'
+        type='button'
+        >
+        <img class="icon" :src="allToAvailable" />
+      </button>
+    </div>
+
+    <div class='chosen'>
+      <label id='chosen-label'>{{ chosenLabel }}</label>
+      <ul
+        aria-labeledby='chosen-label'
+        role='selectbox'
+      >
+        <li
+          v-for="option in chosenOptions"
+          :key="'chosen-option-' + id + '-' + option.id"
```

Comment:
Can we also make these raw strings and not do string concatenation?

Reply:
sure